ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/SyncTargetManager.java

```diff
@@ -59,10 +59,10 @@ public SyncTargetManager(
   }
 
   public CompletableFuture<SyncTarget> findSyncTarget() {
-    return syncState
-        .syncTarget()
-        .map(CompletableFuture::completedFuture) // Return an existing sync target if present
-        .orElseGet(this::selectNewSyncTarget);
+    if (syncState.syncTarget().isPresent() && !syncState.isInSync()) {
+      return CompletableFuture.completedFuture(syncState.syncTarget().get());
```

Comment:
Overall this looks like the right direction.

The `isInSync` check has a small tolerance to avoid rapidly toggling between inSync/outOfSync when a new block is first published but we don't want that behaviour here.  It's common to want different tolerances in different situations so we probably should introduce a `isInSync(long tolerance)` variant and use `isInSync(0)` here.

We could also remove the `syncState.syncTarget().isPresent()` condition because `isInSync` will always be true if there is no sync target.